scipy/sparse/linalg/tests/test_propack.py

```diff
@@ -76,46 +80,89 @@ def test_svdp(ctor, dtype, irl, which):
         check_svdp(n, m, ctor, dtype, k, irl, which)
 
 
-def load_coord(folder, precision, file="illc1850.coord"):
-    dtype = {"single": np.float32, "double": np.float64}[precision]
-    path = os.path.join(folder, precision, file)
-    with open(path) as f:
-        m, n, nnz = (int(val) for val in f.readline().split())
-        coord = np.array([[float(val) for val in line.split()] for line in f])
-    i = coord[:, 0].astype(int) - 1
-    j = coord[:, 1].astype(int) - 1
-    data = coord[:, 2].astype(dtype)
-    A = coo_matrix((data, (i, j)))
+def load_coord(folder, precision, file=None):
```

Comment:
Pretty much everything seems to depend on whether it's real or complex. I would prefer to create separate functions for each - and, if desired, a function that calls one or the other depending on `precision`. (But that logic could also just go in the function `test_examples`.)